pkg/artifact/artifact.go

```diff
@@ -10,13 +10,13 @@ import (
 	"fmt"
 	"io/ioutil"
 	"os"
-	"path"
+	"path/filepath"
 	"regexp"
 	"strings"
 )
 
 var (
-	tarAppNameParse = regexp.MustCompile("([a-zA-Z\\d\\-\\_]+)\\_([a-f0-9]+)\\.tar(\\.gz)?")
+	tarAppNameParse = regexp.MustCompile(`^([\w\-]+)_([a-fA-F0-9]+)\.tar(\.gz)?$`)
```

Comment:
Interestingly, we don't require that the suffix be a hex value - a fact we take advantage of when using `p2-bin2pod` to generate an artifact from a binary (it comes out using the opposite character class, ie `foo_hijzlpohyt.tar.gz`)